docs/tutorial/Tutorial.md

```diff
@@ -0,0 +1,981 @@
+    [2020-12-18T00:27:49Z INFO  mmtk::plan::global]   [POLL] nogc_space: Triggering collection
+    fatal runtime error: failed to initiate panic, error 5
+    Aborted (core dumped)
+    ```
+4. If you haven't already, try building using Semispace. lusearch should now 
+pass, as garbage will be collected, and the smaller benchmarks should run the 
+same as they did while using NoGC.
+
+
+
+### Create MyGC
+NoGC is a GC plan that only allocates memory, and does not have a collector. 
+We're going to use it as a base for building a new garbage collector.
+1. Each plan is stored in `mmtk-openjdk/repos/mmtk-core/src/plan`. Navigate 
+there and create a copy of the folder `nogc`. Rename it to `mygc`.
+3. In *each file* within `mygc`, rename any reference to `nogc` to `mygc`. 
+You will also have to separately rename any reference to `NoGC` to `MyGC`.
+   * For example, in Visual Studio Code, you can (making sure case sensitivity 
+   is selected in the search function) select one instance of `nogc` and either 
+   right click and select "Change all instances" or use the CTRL-F2 shortcut, 
+   and then type `mygc`, and repeat for `NoGC`.
+4. In order to use MyGC, you will need to make some changes to the following 
+files. 
+    1. `mmtk-core/Cargo.toml`, under `#plans`, add: 
+        ```rust
+        mygc = ["immortalspace", "largeobjectspace"]
+        ```
+        This adds a build-time flag for `mygc`, and tells the compiler that 
+        `mygc` will use an immortal space and large object space.
+    2. `mmtk-core/src/plan/mod.rs`, under the import statements, add:
+        ```rust
+        #[cfg(feature = "mygc")]
+        pub mod mygc;
+        #[cfg(feature = "mygc")]
+        pub use self::mygc as selected_plan;
+        ```
+        This adds `mygc` as a module, which can be conditionally compiled 
+        using the feature (or environment variable) `mygc`. A GC plan needs 
+        to be selected at build time, and only one plan can be selected 
+        (as `selected_plan`).
+    3. `mmtk-openjdk/mmtk/Cargo.toml`, under `[features]`, add: 
+        ```rust 
+        mygc = ["mmtk/mygc"] 
+        ```
+        This adds the build flag to the binding crate, using the `mygc` 
+        flag from mmtk-core.
+    
+Note that all of the above changes almost exactly copy the NoGC entries in 
+each of these files. However, NoGC has some variants, such as a lock-free 
+variant, that are not needed for this tutorial. Remove references to them in 
+the MyGC plan now. 
+1. Within `mygc/global.rs`, find any use of `#[cfg(feature = "mygc_lock_free")]` 
+and delete both it *and the line below it*.
+2. Then, delete any use of the above line's negation, 
+`#[cfg(not(feature = "mygc_lock_free"))]`, this time without changing the 
+line below it.
+
+After you rebuild OpenJDK (and the MMTk core), you can use MyGC. Try testing it 
+with the each of the three benchmarks. It should work identically to NoGC.
+
```

Comment:
If you've got to this point, congratulations, you have create your first working MMTk collector!